Week5/Blockbuster/app/src/main/java/com/example/blockbuster/fragments/MainFragment.kt

```diff
@@ -0,0 +1,140 @@
+                }
+            }
+        requireActivity().onBackPressedDispatcher.addCallback(this, callback)
+        //---------------------------------------------------------------------------
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+
+        // Let you inflate an options menu. If you delete the line, the menu isn't show
+        setHasOptionsMenu(true)
+        return inflater.inflate(R.layout.fragment_main, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        //---------------------------------------Show favorite movie
+        val movie = model.data.value?.maxBy { movie -> movie.stars  }
+        if(movie != null){
+            if(movie.poster is Int)
+                iv_fav_poster.setImageResource(movie.poster as Int)
+            else if(movie.poster is Uri)
+                iv_fav_poster.setImageURI(movie.poster as Uri)
+            tv_title_itemf.text = movie.title
+            rb_show_stars.rating = movie.stars
+        }
+        //----------------------------------------------------------
+
+        //---------------------------------Setting the Recycler View
+        val numberOfColumns = 2
+        rv_movies.layoutManager = GridLayoutManager(activity, numberOfColumns)
+        //rv_movies.layoutManager = LinearLayoutManager(activity, LinearLayoutManager.VERTICAL, false)
+        //val adapter = MovieAdapter(movieList as MutableList<Movie>, this)
+        rv_movies.adapter = adapter
+        //----------------------------------------------------------
+
+        // If exist any change on Recycler View's data, then update the view
+        model.data.observe(viewLifecycleOwner, Observer { adapter.updateMovieList(it) })
+
+        // Action for the floating action button
+        fab.setOnClickListener {
+            view.findNavController().navigate(R.id.action_mainFragment_to_addMovieFragment)
+        }
+    }
+
+    override fun listItemClick(movie: Movie) {
+        model.setMovieSelected(movie)
+        view?.findNavController()?.navigate(R.id.action_mainFragment_to_movieFragment)
+    }
+
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.menu_main, menu)
+        // Set the status in the switch and add a listener
+        val item = menu.findItem(R.id.switch_menu)
+        item.setActionView(R.layout.switch_layout)
+        item.actionView.switch_logout.isChecked = model.getLoggedStatus()
+        item.actionView.switch_logout.setOnCheckedChangeListener {_, isChecked ->
+            if (!isChecked)
+                model.setLoggedStatus(false)
+
+            findNavController().navigate(R.id.action_mainFragment_to_loginFragment)
+
+        }
+
+    }
+
+    fun createExitAlert(activity: AppCompatActivity){
```

Comment:
This alert works great, to get the navigation controller to go back to the login activity perhaps use the popBackStack with the destination would work, and perhaps the shared preference value is preventing the user from being able to go back to the login.